kedro-datasets/kedro_datasets/huggingface/hugging_face_dataset.py

```diff
@@ -36,20 +41,53 @@ class HFDataset(AbstractVersionedDataset):
 
     """
 
-    def __init__(self, *, dataset_name: str):
+    def __init__(
+        self,
+        *,
+        dataset_name: str,
+        filepath: str = None,
+        credentials: dict[Any] = None,
+        save_to_disk: bool = True,
+        save_to_hub: bool = False,
+    ):
         self.dataset_name = dataset_name
-
-    def _load(self):
-        return load_dataset(self.dataset_name)
-
-    def _save(self):
-        raise NotImplementedError("Not yet implemented")
+        self.filepath = filepath
+        self.credentials = credentials
+        self.save_to_disk = save_to_disk
+        self.save_to_hub = save_to_hub
+
+    def _load(self) -> Dataset:
+        try:
+            ds = Dataset.load_from_disk(self.filepath)
+        except FileNotFoundError:
+            ds = load_dataset(self.dataset_name)
+        return ds
+
+    def _save(self, data: Dataset):
+        if self.save_to_disk:
+            logger.info("Saving to local disk.")
+            data.save_to_disk(self._filepath)
+
+        if self.save_to_hub:
```

Comment:
I think this branching here is kind of unusual. As an alternative, `kedro-mlflow` provides 2 different datasets for this purpose, `MlflowModelTrackingDataset` and `MlflowModelLocalFileSystemDataset`. Do you think we should do the same here @merelcht ?

Reply:
Yes looking at this it might be an idea to split this into a dataset that does loading & saving to disk and one that does it remotely.